main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintSolver2.scala

```diff
@@ -65,6 +65,14 @@ object ConstraintSolver2 {
       new Soup(newConstrs, moreTree @@ tree)
     }
 
+    /**
+      * Transforms the entire set of constraints with the given many-to-many constraint function.
+      */
+    def blockApply(f: List[TypeConstraint2] => (List[TypeConstraint2], SubstitutionTree)): Soup = {
+      val (newConstrs, moreTree) = f(constrs)
+      new Soup(newConstrs, moreTree @@ tree)
```

Comment:
Worth to check for eq and memory reuse?

Reply:
Probably not. This is a relatively cold piece of code.

I'll see what making a helper does anyway.